commit dd2f61d49d3e2e0e139c1f45c978a492a6a04a80 (refs/changes/16/3716/2)
Author: Nick Zivkovic <nick.zivkovic@joyent.com>
Date:   2018-03-23T23:59:09+00:00 (1 year, 7 months ago)
    
    DOCKER-1114 network inspect of an unknown network should return 404 error

diff --git a/lib/backends/sdc/networks.js b/lib/backends/sdc/networks.js
index 3aba583..40e6533 100644
--- a/lib/backends/sdc/networks.js
+++ b/lib/backends/sdc/networks.js
@@ -468,6 +468,12 @@ function findNetworkOrPoolByNameOrId(name, opts, callback) {
                 getNetworksOrPools(listParams, opts, cb);
             },
             function byDockerId(cb) {
+                if (!name.match(/^[a-f0-9]+$/)) {
+                    log.debug({ name: name },
+                        'Networks: impossible docker id: %s, skipping', name);
+                    setImmediate(cb, null, []);
+                    return;
+                }
                 // we assume the 'double uuid' convention for networks here,
                 // that is, dockerId = (uuid + uuid).replace(/-/g, '').
                 // So if we have a name.length > 31, 32... must be a prefix of
diff --git a/package.json b/package.json
index f0afadd..b3781e9 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "sdc-docker",
-  "version": "0.5.0",
+  "version": "0.5.1",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
diff --git a/test/integration/api-network.test.js b/test/integration/api-network.test.js
index 475ec41..16e81a6 100644
--- a/test/integration/api-network.test.js
+++ b/test/integration/api-network.test.js
@@ -130,4 +130,13 @@ test('docker network inspect', function (tt) {
             t.end();
         });
     });
+    tt.test('inspect non-existent network', function (t) {
+        DOCKER_ALICE.get('/networks/some_random_name',
+            function (err, res, req, net)
+        {
+            t.ok(err, 'expecting error');
+            t.ok(err.statusCode === 404, 'Expecting 404');
+            t.end();
+        });
+    });
 });
